### PR TITLE
Added authentication and token validation

### PR DESCRIPTION
**Pull request by mseheda**
Implemented the ```AuthController``` with a Login method accepting email and hashed password as input params, and generating a JWT for the user.
Added a separate endpoint in AuthController for validating JWT, created a User repository, and secured all other controllers using the ```[Authorize]``` attribute, excluding AuthController. The encryption key is securely stored using the .NET Secrets Tool.

**_Proof of work is provided in the attached screenshots._**

Screenshot 2024-11-29 191602
Screenshot 2024-11-29 191616
Screenshot 2024-11-29 191621
Screenshot 2024-11-29 191640

### CODE REVIEW

File: Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserRepository.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿using Healthcare_Hospital_Management_System.Models;
+
+namespace Healthcare_Hospital_Management_System.Services
+{
+    public class UserRepository : IUserRepository
+    {
+        readonly Dictionary<string, string> users = new Dictionary<string, string>
+        {
+            { "zinchenko.liudmyla@gmail.com", "8b5b9db0c13db24256c829aa364aa90c6d2eba318b9232a4ab9313b954d3555f" },
```
Comment (liudmyla-zinchenko) [inline Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserRepository.cs]: At least change the content of the User Repository, avoid committing user sensitive data

File: Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        private string HashPassword(string plainPassword)
+        {
+            using var sha256 = SHA256.Create();
+            var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(plainPassword));
+            return BitConverter.ToString(hashedBytes).Replace("-", "").ToLower();
+        }
+
+        private string HashPassword(string email, string plainPassword)
```
Comment (liudmyla-zinchenko) [inline Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserService.cs]: Please remove unused method, the method was provided as example of using PasswordHasher

**Review by liudmyla-zinchenko — APPROVED** [review]
Good. Please apply requested changes

File: Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserRepository.cs (hunk 1 above)
Comment (mseheda) [reply Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserRepository.cs]: Removed personal entries, leave only testcases without personal info.

File: Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserService.cs (hunk 2 above)
Comment (mseheda) [reply Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/UserService.cs]: Removed

**Pull request merged**